lib/galaxy/objectstore/__init__.py

```diff
@@ -425,6 +446,79 @@ def _get_concrete_store_description_markdown(self, obj):
     def _get_store_by(self, obj):
         return self.store_by
 
+    @staticmethod
+    def _cache_monitor(*args, **kwargs):
+        stop_cache_monitor_event = args[0]
+        sleeper = args[1]
+
+        time.sleep(kwargs["cache_monitor_startup_delay"])  # Wait for things to load before starting the monitor
```

Comment:
What's that for ? This looks like it's hiding a bug somewhere, we shouldn't propagate this if it isn't needed,

Reply:
This is a 10 year old thing in the code - I'll refactor without it but I think it would fundamentally make sense to refactor with it and then remove it in a separate commit to isolate that change.